File: Server/PromotionServices/models/PromotionModel.js
import mongoose from 'mongoose';

const Schema = mongoose.Schema;

const PromotionSchema = new Schema(
    {
        name: {
            type: String,
            required: true,
        },
        description: {
            type: String,
            required: true,
        },
        discount: {
            type: Number,
            default: 0,
        },
        startDate: {
            type: Date,

            default: Date.now(),
        },
        endDate: {
            type: Date,
            default: Date.now(),
        },
        usageLimit: {
            type: Number,
            default: 0,
        },
    },
    {
        timestamps: true,
    }
);

export const Promotion = mongoose.model('Promotion', PromotionSchema, 'promotions');

File: Client/App/Driver/lib/global/models/promotion/promotion_model.dart
class PromotionModel {
  final String name;
  final String description;
  final double discount;
  final String startDate;
  final String endDate;
  final int usageLimit;

  PromotionModel(
    this.name,
    this.description,
    this.discount,
    this.startDate,
    this.endDate,
    this.usageLimit,
  );

  PromotionModel.fromJson(Map<String, dynamic> json)
      : name = json['name'],
        description = json['description'],
        discount = json['discount'],
        startDate = json['startDate'],
        endDate = json['endDate'],
        usageLimit = json['usageLimit'];
}
